src/buildApplication/checkPackageManager.ts

```diff
@@ -0,0 +1,50 @@
+import YAML from 'js-yaml';
+import { spawn } from 'child_process';
+import { readFile } from 'fs/promises';
+import { validateFile } from '../utils';
+import { cliError } from '../cli';
+
+export async function checkPackageManager() {
```

Comment:
could you please add unit tests for this function? 🙏

Reply:
Hi, I have a problem that I could not unmock the spawn function from the child_process modules, do you know how to solve it?

Reply:
I'm trying to mock it to return an EventEmitter(fake child process) that return 3.0.0 to test it detecting yarn berry, then mock it again to return another EventEmitter that return 1.0.0 to test it detecting yarn classic.
The problem i have is the spawn function still returning 3.0.0 in the yarn classic test

Reply:
nvm, I found a way to fix it